clevertap-core/src/main/java/com/clevertap/android/sdk/LocalDataStore.java

```diff
@@ -454,10 +695,25 @@ private void _setProfileField(String key, Object value) {
      * @param fields, a map of key value pairs to be updated locally. The value will be null if that key needs to be
      *                removed
      */
+//    int k = 0;
     public void updateProfileFields(Map<String, Object> fields) {
         if(fields.isEmpty())
             return;
-
+        /*Set<String> events = new HashSet<>();
+        for (int i = 0; i < 5000; i++) {
+            String s = "profile field - "+k+"-"+i;//RandomStringUtils.randomAlphanumeric(512);
+            events.add(s);
+        }
+        k++;*/
+        long start = System.currentTimeMillis();
+        persistUserEventLogsInBulk(fields.keySet());
+//        persistUserEventLogsInBulk(events);
+        /*for (String key : events)
+        {
+            persistUserEventLog(key);
+        }*/
+        long end = System.currentTimeMillis();
+        config.getLogger().verbose(config.getAccountId(),"UserEventLog: persistUserEventLog execution time = "+(end-start)/1000+" seconds");
```

Comment:
lets print it in ms/ns, i dont think this will take ever time in seconds scale.

Reply:
fixed in https://github.com/CleverTap/clevertap-android-sdk/commit/fcfd5642eb1d183ab607a5013b6a8dabd906e9ab